packages/rs-dpp/src/data_contract/document_type/index/mod.rs

```diff
@@ -188,20 +188,16 @@ impl TryFrom<&[(Value, Value)]> for Index {
         let mut index_properties: Vec<IndexProperty> = Vec::new();
 
         for (key_value, value_value) in index_type_value_map {
-            let key = key_value.to_str().map_err(ProtocolError::ValueError)?;
+            let key = key_value.to_str()?;
 
             match key {
                 "name" => {
                     name = Some(
                         value_value
                             .as_text()
-                            .ok_or({
-                                ProtocolError::DataContractError(
-                                    DataContractError::InvalidContractStructure(
-                                        "index name should be a string".to_string(),
-                                    ),
-                                )
-                            })?
+                            .ok_or(DataContractError::InvalidContractStructure(
```

Comment:
This name of error sounds very ambiguous. Why not to use ValueWrongType?

Reply:
I did not modify this logic I believe.

Reply:
Figuring out what errors we can remove or group together is not the topic of this PR.